### DISCUSSION TITLE
Multi Watch Winder Individually controllable

### DISCUSSION THREAD

**Discussion by neocleous** [mwood77/winderoo#60] [Ideas]
Hello,

Thank you for all of your development time on this, I put the project together and it's great.

In my mind I imagine having 8 of these in a box like a high-end watch winder but with all the smart functionality offered by the ESP32.

Would it possible to have different Winderoo's selectable in the UI? 
I imagine DNS may be an issue here because each instance wants to resolve to winderoo.local 
Also, would it be possible to label each winderoo for a watch? For example, on the OLED screen give it a user editable name like Rolex to identify the watch.

Thanks again for this great project

**Comment by mwood77** [comment]
Thanks for your kind words - your post slipped through my notifications somehow.

I've been slowly poking away at this kind of functionality, however I've been using ESP-NOW (mesh) instead of DNS. Idea being that one devices becomes the main device which handles DNS resolution, while the others drop off the network and only communicate over the mesh.

I've had a couple hiccups sorting this out, as the communication drops off or becomes unreachable after the initial handshake between the main and secondary winder.

You can see the draft work here: https://github.com/mwood77/winderoo/pull/50/files

The biggest challenge is handling the mesh messages in an "almost asynchronous" manner; the internal watchdog will kill the web server's process if it doesn't return in time. That's mostly sorted now, but I'm "stuck" getting the units to communicate reliably to each other.

Regarding the UI and naming - adding an addressable name on the OLED should be a piece of cake; rebuilding the web GUI to handle multiple winders will be a bit more nuanced. However, the API can already inform the web GUI about the presence of multiple winders on the mesh network.

----

If you're comfortable fiddling with the code, you can get around the DNS issue and run multiple winders on your network right now. The caveat being that you'll need to rename the host of each winder individually.

So it would be like:
- winder1.local
- winder2.local
- ...

I mention how to do that here:

https://github.com/mwood77/winderoo/issues/8

**Reply by neocleous** [reply]
Thanks for your detailed response, I'll put together another watch winder and give it a go.

It makes sense having one run as a master and the others connecting back like a lot IoT devices saves a lot of network traffic. Perhaps using some sort of dedicated hub would help on that side of it? 

If the devices are close together in a larger watch winder box could they be electrically connected through the GPIO pins rather than run wirelessly?

I was planning to add a Raspberry Pi running a web client in a kiosk mode with a touch screen as the control unit. I think I'll go ahead and purchase the parts as a POC.

Regarding the motor selection in the BOM, as I mentioned on Printables the MX1508 gave a quitter experience with the motor, but it was still quite loud so I tried a [different motor](https://www.bastelgarage.ch/6v-40rpm-10kg-cm-dc-metal-gear-worm-motor?search=6v%2040rpm%2010kg%2Acm%20dc%20metal%20gear%20worm%20motor) with a worm drive gear box and this was much quitter. There are plenty of these on AliExpress but I didn't want to wait, next time i'Il order it will be from AliExpress. 

It has the added benefit of the drive shaft being at 90 degrees to the motor with an M3 thread in the drive shaft meaning direct drive could be possible. I would imagine doing that with the code as it is now would throw off turns per day due to the pullies being accounted for in the current code.

Thanks again for your work on the project.

**Reply by neocleous** [reply]
I took a look at one of the remix models that replace the pullies with gears and I calculated that the effective RPM of the watch winder is 23 RPM, there is a worm drive motor that exactly matches that [speed.](https://www.aliexpress.com/item/1005005918560507.html?spm=a2g0o.productlist.main.45.87fdRpYORpYOES&algo_pvid=6c25d7ea-9a38-4693-bc9c-c73465223697&algo_exp_id=6c25d7ea-9a38-4693-bc9c-c73465223697-22&pdp_npi=4%40dis%21CHF%2115.81%2110.28%21%21%2117.69%2111.50%21%40210384db17225046300672199eac5e%2112000034851922215%21sea%21CH%21733719614%21X&curPageLogUid=psGKStKxfVHZ&utparam-url=scene%3Asearch%7Cquery_from%3A)

**Reply by mwood77** [reply]
Excellent find!

------------

> If the devices are close together in a larger watch winder box could they be electrically connected through the GPIO pins rather than run wirelessly?

I guess, although I hadn't thought about physically connecting them. I can imagine that it would complicate the construction significantly though.

> It has the added benefit of the drive shaft being at 90 degrees to the motor with an M3 thread in the drive shaft meaning direct drive could be possible. I would imagine doing that with the code as it is now would throw off turns per day due to the pullies being accounted for in the current code.

Since you're using an MX1508, you can adjust the rotation speed here:
https://github.com/mwood77/winderoo/blob/b2d59a803a1c96bf578acefc24e2de71b626fc6e/src/platformio/osww-server/src/utils/MotorControl.cpp#L7

When that's rotating at a speed you're happy with, update this value here:
https://github.com/mwood77/winderoo/blob/b2d59a803a1c96bf578acefc24e2de71b626fc6e/src/platformio/osww-server/src/main.cpp#L38

That way, the timing, cycle completion time, and other values continue to be in-sync.
If your timing is different, you can adjust that value.

**Reply by neocleous** [reply]
I think there are a few ways at looking at multiple watch winders. I would imagine that they will either be in a purpose made box or created individually and kept together or kept fully independant which is where the mesh idea you have been working on makes a lot of sense.

To keep the interconnect simple, [magnetic pogo pins](https://www.aliexpress.com/w/wholesale-magnetic-pogo-pins.html?spm=a2g0o.detail.search.0) could be used in combination with the clips you created. That would remove the complexity of wiring them together as the number of watch winders expands.

Obviously in a box jumper cables would do the job
